plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java

```diff
@@ -539,9 +539,7 @@ private void createOrUpdateAssetAndContents(final StorageTx storageTx, final Buc
 
   private void attachBlob(final StorageTx storageTx, final Component component, final Asset asset, final InputStream in)
   {
-    final ImmutableMap<String, String> headers = ImmutableMap
-        .of(BlobStore.BLOB_NAME_HEADER, blobName(component), BlobStore.CREATED_BY_HEADER, "unknown");
-    storageTx.setBlob(in, headers, asset, singletonList(HashAlgorithm.SHA512), "application/zip");
+    storageTx.setBlob(asset, asset.name(), in, singletonList(HashAlgorithm.SHA512), null, "application/zip");
```

Comment:
Previously the blob name was `blobName(component)` which included the version (and repo) whereas now it's `asset.name()` which is just the components NuGet id without version. None of these names include the nominal NuGet extension of `.nupkg`.

We should be storing both id and version in the blob name, so it should go back to using `blobName(component)` and the implementation of `blobName` should probably be altered to create a name like "{id}-{version}.nupkg"?